solution/parser/ecfr-parser/parsexml/postprocessing_test.go

```diff
@@ -5,85 +5,331 @@ import (
+				Citation: []string{},
+				Marker: []string{},
+			},
+			Previous: &Paragraph{
+				Type: "Paragraph",
+				Content: "(3) <I>Chronological order by bill submission date.</I> Under this option, the agency...",
+				Citation: []string{"h", "3"},
+				Marker: []string{"3"},
+			},
+			Expected: []string{"i", "1"},
+			Error: false,
+		},
+		{
+			Name: "test-letter-vs-roman-i",
+			Paragraph: &Paragraph{
+				Type: "Paragraph",
+				Content: "(i) The agency must execute written agreements with other agencies before releasing...",
+				Citation: []string{},
+				Marker: []string{},
+			},
+			Previous: &Paragraph{
+				Type: "Paragraph",
+				Content: "(h) Information exchanged electronically between the State Medicaid agency and any...",
+				Citation: []string{"h"},
+				Marker: []string{"h"},
+			},
+			Expected: []string{"i"},
+			Error: false,
+		},
+		{
+			Name: "test-letter-vs-roman-v-1",
+			Paragraph: &Paragraph{
+				Type: "Paragraph",
+				Content: "(v) <I>Applicability date.</I> Sections 438.3(h) and (q) apply to the rating...",
+				Citation: []string{},
+				Marker: []string{},
+			},
+			Previous: &Paragraph{
+				Type: "Paragraph",
+				Content: "(u) <I>Recordkeeping requirements.</I> MCOs, PIHPs, and PAHPs must retain, and...",
+				Citation: []string{"u"},
+				Marker: []string{"u"},
+			},
+			Expected: []string{"v"},
+			Error: false,
+		},
+		{
+			Name: "test-letter-vs-roman-v-2",
+			Paragraph: &Paragraph{
+				Type: "Paragraph",
+				Content: "(v) <I>Applicability date.</I> Sections 438.3(h) and (q) apply to the rating...",
+				Citation: []string{},
+				Marker: []string{},
+			},
+			Previous: &Paragraph{
+				Type: "Paragraph",
+				Content: "(u) <I>Recordkeeping requirements.</I> MCOs, PIHPs, and PAHPs must retain, and...",
+				Citation: []string{"u", "v", "w"},
+				Marker: []string{"u"},
+			},
+			Expected: []string{"v"},
+			Error: false,
+		},
+		{
+			Name: "test-wrong-paragraph-order",
+			Paragraph: &Paragraph{
+				Type: "Paragraph",
+				Content: "(iv) Individuals whose eligibility was determined under a State's option under...",
+				Citation: []string{},
+				Marker: []string{},
+			},
+			Previous: &Paragraph{
+				Type: "Paragraph",
+				Content: "(b) <I>Negative case reviews.</I> Except as provided in paragraph (c) of this...",
+				Citation: []string{"b"},
+				Marker: []string{"b"},
+			},
+			Expected: nil,
+			Error: true,
+		},
+	}
+
+	for _, tc := range testTable {
+		t.Run(tc.Name, func(t *testing.T) {
+			out, err := generateParagraphCitation(tc.Paragraph, tc.Previous)
+			if err != nil && !tc.Error {
+				t.Errorf("expected no error, received (%+v)", err)
+			} else if err == nil && tc.Error {
+				t.Errorf("expected error, received (%+v)", out)
+			} else if err == nil && !reflect.DeepEqual(out, tc.Expected) {
+				t.Errorf("expected (%+v), received (%+v)", tc.Expected, out)
+			}
+		})
+	}
+}
+
+func TestMatchLabelType(t *testing.T) {
```

Comment:
This is out of scope for the test, but I don't see this using values above 1 in the code.